service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamPermissionInspector.java

```diff
@@ -58,12 +54,23 @@ public boolean checkCachedPermission(String accessToken) {
     return expiresAt.isAfter(Instant.now());
   }
 
-  protected Optional<String> getToken(String cookieValue) {
+  protected Optional<String> getToken(Map<String, String> headers) {
+    return getTokenFromCookie(headers).or(() -> getTokenFromAuthorization(headers));
```

Comment:
This means the check order would be cookie and then Auth header. I think the reverse may be better. The Auth header is the standard way of checking out. The cookie should be the fallback.  How does Leo currently do it?